### ISSUE TITLE
Download/upload settings removes some plugins

### ISSUE THREAD

**Issue by gyandeeps** [shanalikhan/code-settings-sync#482]
**Visual Studio Code Version :** 
```
Version 1.19.2
Commit 490ef761b76b3f3b3832eff7a588aac891e5fe80
Date 2018-01-10T15:55:03.538Z
Shell 1.7.9
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64
```
**Code Settings Sync Version :** 2.8.7 (latest)
**Operating System :** Windows7/10
**Occurs On:** Upload / Download 
**Proxy Enabled:** No
**GIst Id:** https://gist.github.com/gyandeeps/5e618c3e92cb56cb73b20dbe2719cea3

### Sync settings used

```
    "sync.autoDownload": true,
    "sync.autoUpload": true,
    "sync.lastDownload": "2018-01-12T18:19:07.367Z",
    "sync.forceDownload": false,
    "sync.anonymousGist": false,
    "sync.host": "",
    "sync.pathPrefix": "",
    "sync.quietSync": false,
    "sync.askGistName": false,
```

### Steps

* plugin installed on  machine M1 and M2
* Both have extension E1 and E2
* Go to M2 and install E3
* Open to M1
  * Download new settings and start installing new plugins 
  * In parallel upload settings where E3 is removed.
  * Check my gist for more info

### Expectation

* E3 should not be removed.

 
I am willing to help to fix this issue. I would really appreciate some sort of direction on where to start looking.

**Comment by shanalikhan** [comment]
Sorry i think im little confused with the steps.
 
When you install E3 in M2 you need to upload the settings and then download the settings on M1, which will install E3 and will remove all the extension in M1 that are not in M2 ( as they are not uploaded)

**Comment by gyandeeps** [comment]
my sync settings are
```
    "sync.autoDownload": true,
    "sync.autoUpload": true,
```

So it will auto download and upload.

**Comment by nomoresecrets** [comment]
Having the same issue, kind of annoying when syncing between four machines. Seems only to happen when autoupload *and* autodownload is active

**Comment by X-Tender** [comment]
I have autoupload and autodownload setted to false but the plugins get still removed.
An option which tell not to remove already installed plugins would be nice. :)

**Comment by shanalikhan** [comment]
Duplicate of #396

**Comment by shanalikhan** [comment]
>I am willing to help to fix this issue. I would really appreciate some sort of direction on where to start looking.

@gyandeeps This issue is a duplicate of other issue i have mentioned, I am open for Pull request and help you do so, if you are willing.

> An option which tell not to remove already installed plugins would be nice. :)

@X-Tender check this out, https://github.com/shanalikhan/code-settings-sync/issues/389 - will work on it.

**Comment by Mellbourn** [comment]
I have keep having a similar problem: I have two machines, PC and Mac, with Code and Settings Sync.

1. I install an extension on mac
2. It syncs
3. I start Code on the PC
4. It syncs, *and removes the extension*
5. I go back to my mac
6. It syncs, *and removes the extension*
7. I go back to PC
8. It syncs, and installs the extension!?
9. going back and forth, the two machines keep installing and uninstalling the extension, as if they were fighting.

This has happened to me several times for different extensions. Unfortunately it makes me not trust the syncing.

```JavaScript
  "sync.autoDownload": true,
  "sync.autoUpload": true,
  "sync.forceDownload": false,
```

Settings Sync: 2.8.7
Code: 1.20.1
OS: macOS, Windows 10

My settings are here https://gist.github.com/Mellbourn/2cb4e0da308c4bb3b086d64370238de7 the scenario in question happened on 2018-02-24 regarding the "git indicators" extension.

**Comment by peterschussheim** [comment]
I have been fighting the exact same behavior that @Mellbourn posted.  @shanalikhan please, please, please provide an update on **why** this is happening and what is being done to fix this issue.

**Comment by shanalikhan** [comment]
@peterschussheim this happens with the auto upload and download enable.
I'm the only one adding the features and maintaining. It will take some time to fix those issues.
Also I'm open for the Pull request and will release new version if anyone solve it.
